python/cusignal/_upfirdn.py

```diff
@@ -197,66 +262,184 @@ def _init_raw_apply1d_modules():
+        module.get_function("_cupy_upfirdn_1d_float")
+    _cached_modules['_cupy_upfirdn_1d_double'] = \
+        module.get_function("_cupy_upfirdn_1d_double")
+    _cached_modules['_cupy_upfirdn_1d_complex_float'] = \
+        module.get_function("_cupy_upfirdn_1d_complex_float")
+    _cached_modules['_cupy_upfirdn_1d_complex_double'] = \
+        module.get_function("_cupy_upfirdn_1d_complex_double")
+
+@cp.prof.TimeRangeDecorator()
+def _cupy_init(blockspergrid, threadsperblock, x, h_trans_flip, up, down, axis, stream, out):
+
+    x_shape_a = x.shape[axis]
+    h_per_phase = len(h_trans_flip) // up
+    padded_len = x.shape[axis] + h_per_phase - 1
+
+    if out.ndim > 1:
+        raise NotImplementedError(
+            "Raw CuPy Kernel is not implemented \
+            for ndim > 1"
+                )
+    else:
+        _init_cupy_upfirdn_1d_modules()
+        cupy_upfirdn_1d_float = _cached_modules['_cupy_upfirdn_1d_float'] # 21 registers - 9-12us
+        cupy_upfirdn_1d_double = _cached_modules['_cupy_upfirdn_1d_double'] # 30 registers - 14us
+        cupy_upfirdn_1d_complex_float = _cached_modules['_cupy_upfirdn_1d_complex_float'] # 32 registers - 22us
+        cupy_upfirdn_1d_complex_double = _cached_modules['_cupy_upfirdn_1d_complex_double'] # 34 registers - 50-70us
+
+        stream.use()
+
+        if out.dtype == cp.float32:
+            cupy_upfirdn_1d_float(
+                (blockspergrid,),
+                (threadsperblock,),
+                (out.shape[0], 
+                x,
+                h_trans_flip, 
+                up, 
+                down,
+                x_shape_a, 
+                h_per_phase, 
+                padded_len,
+                out,),
+            )
+        elif out.dtype == cp.float64:
+            cupy_upfirdn_1d_double(
+                (blockspergrid,),
+                (threadsperblock,),
+                (out.shape[0], 
+                x,
+                h_trans_flip, 
+                up, 
+                down,
+                x_shape_a, 
+                h_per_phase, 
+                padded_len,
+                out,),
+            )
+
+        elif out.dtype == cp.complex64:
+            cupy_upfirdn_1d_complex_float(
+                (blockspergrid,),
+                (threadsperblock,),
+                (out.shape[0], 
+                x,
+                h_trans_flip, 
+                up, 
+                down,
+                x_shape_a, 
+                h_per_phase, 
+                padded_len,
+                out,),
+            )
+
+        elif out.dtype == cp.complex128:
+            cupy_upfirdn_1d_complex_double(
+                (blockspergrid,),
+                (threadsperblock,),
+                (out.shape[0], 
+                x,
+                h_trans_flip, 
+                up, 
+                down,
+                x_shape_a, 
+                h_per_phase, 
+                padded_len,
+                out,),
+            )
```

Comment:
Let's add an else and raise a NotImplementedError if dtype isn't complex64/128 or float32/64

Reply:
Does this wording work?
```
NotImplementedError: Out type - dtype('int32') not supported
```